app/services/blog_search.rb

```diff
@@ -0,0 +1,10 @@
+class BlogSearch
+	include ActionView::Helpers::SanitizeHelper
+	include ActiveModel::Model
+
+	attr_accessor :contents
+
+	def search
+		Post.left_joins(:tags).joins(:blog).where("lower(blogs.title) like ? or lower(posts.title) like ? or lower(posts.body) like ? or lower(tags.name) like ?", "%#{contents.downcase}%", "%#{contents.downcase}%", "%#{contents.downcase}%", "%#{contents.downcase}%").uniq
+	end
```

Comment:
Interesting to see how you'd approach it. Thanks!

Reply:
@JustinJianQiuLi I'd be interested in knowing (via comment) how you'd approach this for a larger data set (e.g. hundreds of users with multiple blogs, each with hundreds of posts). `like` and `or` queries get very slow after a time.

Not sure if you're aware, you can use named parameters in queries as well, so you only need to pass the value in once:
```ruby
where("LOWER(blogs.title) LIKE :contents OR LOWER(posts.title) LIKE :contents ...", contents: "%#{contents.downcase}%")
```
(You can also use `Post.select('DISTINCT *')...` instead of `.uniq` to return unique results which returns it as an `ActiveRecord::Relation` rather than being converted to an array.)

Reply:
@mattr Thanks for your reply! I am very happy to know this, it is really helpful. 

If the data set become large, I prefer to separate those queries to avoid "or". each query get those posts ids, and finally use where(id: ids) to get the final results.